### ISSUE TITLE
java.lang.ClassCastException: class org.apache.avro.LogicalTypes$Date cannot be cast to class org.apache.avro.Date 

### ISSUE THREAD

**Issue by cesarbruschetta** [zolyfarkas/avro#9]
When loading a json that in its schema uses [logical type Date](https://avro.apache.org/docs/current/spec.html#Date), the schema below

```
{
  "type": "int",
  "logicalType": "date"
}
```

 I get the error

`Caused by: java.lang.ClassCastException: class org.apache.avro.LogicalTypes$Date cannot be cast to class org.apache.avro.Date (org.apache.avro.LogicalTypes$Date and org.apache.avro.Date are in unnamed module of loader 'app')`

lowered I'll put the complete traceback

```
org.apache.avro.AvroRuntimeException: Cannot convert 19073:Integer: expected generic type
        at org.apache.avro.Conversions.convertToLogicalType(Conversions.java:259)
        at org.apache.avro.generic.GenericDatumReader.convert(GenericDatumReader.java:195)
        at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:134)
        at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:152)
        at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:138)
        at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:223)
        at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:213)
        at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:144)
        at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:138)
        at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:115)
        at com.grupozap.validator.validation.JsonToAvroDeserializer.validate(JsonToAvroDeserializer.java:27)
        at com.grupozap.validator.validation.JsonToAvroDeserializer.validateRecord(JsonToAvroDeserializer.java:61)
        at com.grupozap.validator.service.ValidatorService.valid(ValidatorService.java:23)
        at com.grupozap.validator.controller.ValidatorController.post(ValidatorController.java:22)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
        at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
        at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:879)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:793)
        at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:660)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:373)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1594)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassCastException: class org.apache.avro.LogicalTypes$Date cannot be cast to class org.apache.avro.Date (org.apache.avro.LogicalTypes$Date and org.apache.avro.Date are in unnamed module of loader 'app')
        at org.apache.avro.data.DateConverter.fromInt(DateConverter.java:91)
        at org.apache.avro.data.DateConverter.fromInt(DateConverter.java:17)
        at org.apache.avro.Conversions.convertToLogicalType(Conversions.java:245)
```

**Comment by zolyfarkas** [comment]
Can you let me know what version of the avro library are you using?

**Comment by zolyfarkas** [comment]
I think I found an issue, pushing out a new version with a fix: 1.10.0.9p
